common/persistence/visibility/store/query/converter.go

```diff
@@ -479,24 +479,36 @@ func (c *comparisonExprConverter) Convert(expr sqlparser.Expr) (elastic.Query, e
 
 	var query elastic.Query
 	switch comparisonExpr.Operator {
-	case ">=":
+	case sqlparser.GreaterEqualStr:
 		query = elastic.NewRangeQuery(colName).Gte(colValues[0])
-	case "<=":
+	case sqlparser.LessEqualStr:
 		query = elastic.NewRangeQuery(colName).Lte(colValues[0])
-	case ">":
+	case sqlparser.GreaterThanStr:
 		query = elastic.NewRangeQuery(colName).Gt(colValues[0])
-	case "<":
+	case sqlparser.LessThanStr:
 		query = elastic.NewRangeQuery(colName).Lt(colValues[0])
-	case "=", "like": // The only difference is that "%" is removed for LIKE queries.
+	case sqlparser.EqualStr, sqlparser.LikeStr: // The only difference is that "%" is removed for LIKE queries.
 		// Not elastic.NewTermQuery to support partial word match for String custom search attributes.
 		query = elastic.NewMatchQuery(colName, colValues[0])
-	case "!=", "not like":
+	case sqlparser.NotEqualStr, sqlparser.NotLikeStr:
 		// Not elastic.NewTermQuery to support partial word match for String custom search attributes.
 		query = elastic.NewBoolQuery().MustNot(elastic.NewMatchQuery(colName, colValues[0]))
-	case "in":
+	case sqlparser.InStr:
 		query = elastic.NewTermsQuery(colName, colValues...)
-	case "not in":
+	case sqlparser.NotInStr:
 		query = elastic.NewBoolQuery().MustNot(elastic.NewTermsQuery(colName, colValues...))
+	case sqlparser.StartsWithStr:
+		v, ok := colValues[0].(string)
+		if !ok {
+			return nil, NewConverterError("right-hand side of '%v' must be a string", comparisonExpr.Operator)
+		}
+		query = elastic.NewPrefixQuery(colName, v)
+	case sqlparser.NotStartsWithStr:
+		v, ok := colValues[0].(string)
+		if !ok {
+			return nil, NewConverterError("right-hand side of '%v' must be a string", comparisonExpr.Operator)
+		}
+		query = elastic.NewBoolQuery().MustNot(elastic.NewPrefixQuery(colName, v))
```

Comment:
Don't you need to enable [index_prefixes](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-prefixes.html) for this to work?

Reply:
No. index_prefixes would make search faster, but not required.